cookbooks/bcpc/libraries/utils.rb

```diff
@@ -67,11 +78,26 @@ def make_config(key, value)
 	end
 end
 
-def get_config(key)
-	init_config if $dbi.nil?
-	Chef::Log.info  "------------ Fetching value for key \"#{key}\""
-	value = node['bcpc']['encrypt_data_bag'] ? $edbi[key] : $dbi[key]
-	return value
+# get value for data bag/chef-vault item with key
+# bag: databag name of chef vault item; 
+# item: chef vault item; 
+# key: the key to retrieve password in the item
+def get_config(bag="configs",item=node.chef_environment,key)
```

Comment:
Shouldn't the order for parameters be (key, item, bag)? A `bag` can have multiple `items` and an `item` can have multiple `keys`. Consider a scenario where default bag `configs` has two items and I want to fetch a particular `key` from a particular `item` leaving the data bag to default value. The call the get_config method will look something like:
````
get_config(item,key)
````

With current method signature the call will change the bag to the item name passed instead of changing the item parameter. Refer to the code below:

````
def get_config(bag="configs",item="chef_environment",key)
  puts "Bag is : #{bag}"
  puts "Item is : #{item}"
  puts "Key is : #{key}"
end
get_config("myitem","test")
````
The output from above I get is:
````
Bag is : myitem
Item is : chef_environment
Key is : test
````